This notebook is part of the Coursera, IBM Advanced Data Science Capstone Project

Data used in this Capstone Project is found in UCI Machine Learning Repisotory on kaggle website titled as "Air Pressure System Failures in Scania Trucks" (https://www.kaggle.com/uciml/aps-failure-at-scania-trucks-data-set#aps_failure_training_set_processed_8bit.csv)

It is a structured, relational, tabular data in csv format.
It is not streaming (persistent) data.

This notebook is for the "Extract Transform Load (ETL)" stage of the Process Model used for this data science project.

Deliverable of this notebook is the data to be used in other stages of the process.


In [1]:
# Importing pandas for working with the dataframe

import pandas as pd
import numpy as np

In [2]:
# The code was removed by Watson Studio for sharing.

In [20]:
df.head()

class    aa_000    ab_000    ac_000    ad_000    ae_000    af_000  \
0 -0.992188  0.117188 -0.289062  0.992188 -0.007812 -0.046875 -0.054688   
1 -0.992188 -0.179688 -0.289062 -0.468750 -0.007812 -0.046875 -0.054688   
2 -0.992188 -0.125000 -0.289062 -0.468750 -0.007812 -0.046875 -0.054688   
3 -0.992188 -0.406250 -0.289062 -0.468750 -0.007812 -0.046875 -0.007812   
4 -0.992188  0.007812 -0.289062 -0.468750 -0.007812 -0.046875 -0.054688   

     ag_000   ag_001    ag_002    ...       ee_002    ee_003    ee_004  \
0 -0.007812 -0.03125 -0.054688    ...     0.687500  0.515625  0.234375   
1 -0.007812 -0.03125 -0.054688    ...    -0.023438 -0.062500 -0.132812   
2 -0.007812 -0.03125 -0.054688    ...    -0.140625 -0.093750 -0.015625   
3 -0.007812 -0.03125 -0.054688    ...    -0.382812 -0.382812 -0.375000   
4 -0.007812 -0.03125 -0.054688    ...     0.156250  0.031250 -0.031250   

     ee_005    ee_006    ee_007    ee_008    ee_009    ef_000    eg_000  
0  0.070312  0.007812 -0.109375 -0.140625 -0.171875 -0.023438 -0.023438  
1 -0.132812 -0.187500 -0.148438 -0.085938 -0.140625 -0.023438 -0.023438  
2  0.015625 -0.007812 -0.109375 -0.093750 -0.164062 -0.023438 -0.023438  
3 -0.351562 -0.312500 -0.195312 -0.304688 -0.171875  0.890625  0.992188  
4 -0.039062 -0.046875 -0.015625  0.656250 -0.148438 -0.023438 -0.023438  

[5 rows x 171 columns]

In [23]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60000 entries, 0 to 59999
Columns: 171 entries, class to eg_000
dtypes: float64(171)
memory usage: 78.3 MB


We have 171 features (all float data type) and a class column (also float data type, we need to make it categorical at the following stage)

In [25]:
df.isnull().sum().sum()

0

We have no na values within the dataframe

Dataframe can be used in other stages of the project without any transformation